webpack/assets/javascripts/react_app/components/EditableTextDisplay/EditableTextDisplay.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Spinner,
+  TextInput,
+  TextArea,
+  Text,
+  TextVariants,
+  Button,
+  Split,
+  SplitItem,
+} from '@patternfly/react-core';
+import { TimesIcon, CheckIcon, PencilAltIcon } from '@patternfly/react-icons';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../common/I18n';
+import './editableTextDisplay.scss';
+
+const EditableTextDisplay = ({ onEdit, value, textArea, attribute }) => {
+  // Tracks input box state
+  const [inputValue, setInputValue] = useState(value);
+  const [editing, setEditing] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  // Setting didCancel to avoid actions from happening after component has been unmounted
+  // see https://overreacted.io/a-complete-guide-to-useeffect/#speaking-of-race-conditions
+  useEffect(() => {
+    let didCancel = false;
+
+    const onSubmit = async () => {
+      if (submitting) {
+        await onEdit(inputValue, attribute);
+        if (!didCancel) {
+          setSubmitting(false);
+          setEditing(false);
+        }
+      }
+    };
+    onSubmit();
+
+    return () => {
+      didCancel = true;
+    };
+  }, [submitting]);
+
+  // Listen for enter and trigger submit workflow on enter
+  useEffect(() => {
+    const listener = event => {
+      if (event.code === 'Enter' || event.code === 'NumpadEnter') {
+        if (editing) setSubmitting(true);
+      }
+    };
+    document.addEventListener('keydown', listener);
+    return () => {
+      document.removeEventListener('keydown', listener);
+    };
+  }, [editing]);
+
+  const onClear = () => {
+    setInputValue(value);
+    setEditing(false);
+  };
+
+  const inputProps = {
+    value: inputValue || '',
+    onChange: v => setInputValue(v),
+  };
+
+  if (submitting) return <Spinner size="sm" />;
+  if (editing) {
+    return (
+      <Split>
+        <SplitItem>
+          {textArea ? (
+            <TextArea {...inputProps} aria-label={`${attribute} text area`} />
+          ) : (
+            <TextInput
+              {...inputProps}
+              type="text"
+              aria-label={`${attribute} text input`}
+            />
+          )}
+        </SplitItem>
+        <SplitItem>
+          <Button
+            aria-label={`submit ${attribute}`}
+            variant="plain"
+            onClick={() => setSubmitting(true)}
+          >
```

Comment:
As I mentioned above - 
```suggestion
          <Button
            aria-label={`submit ${attribute}`}
            variant="plain"
            onClick={() => onSubmit}
          >
```